### PR TITLE
Fix: uses the new profile name when building & pushing images

### LINKED ISSUE #180: Problem with profile name when using role-arn to build and push the image
### Orb version
8.0.0

### What happened

This version has a problem with role-arn assumption,
When we don't pass "new-profile-name" but we want to assume a role-arn, it sets empty string "" as profile-name for the role assumption step so it gets failed,

If we pass the "new-profile-name" the role assumption part is working but the rest of the steps still using the "default" value for the profile-name, so they don't use the assumed credentials to do the job,

The temporary solution is to just set "default" as value for "new-profile-value", or set same name for all three parameters "new-profile-name", "profile-name" and "source-profile"

### Expected behaviour

Instead of having a parameter named "new-profile-name" I'd suggest having only two parameters "profile-name" and "source-profile" and "source-profile for before assuming the role and" after assuming the role-arn using the profile-name for the next steps.
 
Or if you want to keep the new-profile-name parameter, use new-profile-name for the steps after the role assumption instead of the profile-name would help

### PR DESCRIPTION
**Pull request by kelvintaywl**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [ ] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Description

As noted in an earlier issue @ #180, I was also unable to utilize the `build-and-push-image` job when setting up with `role-arn` and `new-profile-name` parameters.

Specifically, I was seeing that the `aws ecr ger-login-password` failing due to it [using the `profile-name` rather than `new-profile-name` (which was set up with the supplied role ARN)](https://app.circleci.com/pipelines/github/kelvintaywl-cci/aws-ecr-orb-explore/1/workflows/9fe80e75-b9db-4ae3-9af6-e7e282bc9c12/jobs/2?invite=true#step-105-24).
```
An error occurred (AccessDeniedException) when calling the GetAuthorizationToken operation: User: arn:aws:iam::123456789012:user/kelvin-assume-role-only is not authorized to perform: ecr:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr:GetAuthorizationToken action
```

In this PR, I made a fix on the following commands/jobs to ensure the `new-profile-name` param is used for AWS profiles when `role-arn` and `new-profile-name` is set.

- jobs.build-and-push-image
- commands.ecr-login

With this fix coming from my fork of this repo, you can see this updated version imported and working successfully in a test repo here:

https://app.circleci.com/pipelines/github/kelvintaywl-cci/aws-ecr-orb-explore/13/workflows/6bf0a45a-e42b-4c3b-8d9d-aeeb8c01c1d0/jobs/14

Hopefully, this should also resolves #180

### CODE REVIEW

File: src/jobs/build-and-push-image.yml [hunk 1]
```diff
@@ -233,19 +249,43 @@ steps:
...
-      extra-build-args: <<parameters.extra-build-args>>
-      no-output-timeout: <<parameters.no-output-timeout>>
-      skip-when-tags-exist: <<parameters.skip-when-tags-exist>>
-      profile-name: <<parameters.profile-name>>
-      platform: <<parameters.platform>>
-      region: <<parameters.region>>
-      public-registry: <<parameters.public-registry>>
-      push-image: <<parameters.push-image>>
-  - store_artifacts:
-      path: test.txt
```
Comment (kelvintaywl) [inline src/jobs/build-and-push-image.yml:251]: i understand this was likely erroneously committed, hence i removed it in this PR.

**Review by KyleTryon — COMMENTED** [review]
Just want to confirm with @brivu if this is along the lines of what he was originally envisioning when making the latest major update, or if there is maybe an alternative solution already in place.

**Comment by kelvintaywl** [comment]
For sharing,

I did consider a possible solution of setting the `AWS_PROFILE` env var.
However, I noted from the AWS documentation that in situations where the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars are set, that takes precedence over this `AWS_PROFILE` env var.

Hence, given that many users are likely to have the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars in the project (belonging to a IAM user), I decided against this approach.

**Comment by brivu** [comment]
Hey Kelvin!

Thanks for this PR! I'd love to explore this more with you!

In order to use the `role-arn` feature, the user needs to: 

1. Configure a profile (defaults to `default`) with the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` using the `aws-cli/setup` command
2. When assuming the a new role, `step 1` must be run first. The `aws-cli/role-arn-setup` command takes the `role-arn`, `new-profile` and `source-profile` as parameters so that the `new-profile` inherits the credentials of the `source-profile`.  

The idea was to let users configure the `role-arn` and assume the role using the `aws sts assume-role` command after configuration.  Here's an example:

``` 
  jobs:
    configure-role-arn:
      executor: aws-cli/default
      steps:
        - checkout
        - aws-cli/setup: #  Required: Run setup command first to configure credentials for source profile
            profile-name: default
        - aws-cli/role-arn-setup:
            profile-name: new-profile
            role-arn: "arn:aws:iam::123456789012:role/example-role"
            source-profile: default # Use profile created from setup command
        - run: aws sts assume-role --role-arn "arn:aws:iam::123456789012:role/example-role" --role-session-name AWSCLI-Session
```
However, what this did not account for was enabling the user to use the newly configured profile in other commands in this orb, specifically like below, correct?

```
aws-ecr/build-and-push-image:
          name: integration-tests-named-profile
          attach-workspace: true
          profile-name: NEW-PROFILE-NAME
```

I'm wondering if when the `role-arn` is configured, we can create an environment variable in the bash script that can be accessible to the user later on. For example if we can do `export ARN_PROFILE="${PARAM_AWS_CLI_PROFILE_NAME}"` in the `aws-cli/role-arn-setup` command, we might be able to allow users to use this profile in other jobs like so:

```
aws-ecr/build-and-push-image:
          name: integration-tests-named-profile
          attach-workspace: true
          profile-name: ${ARN_PROFILE}
```

@felicianotech, @KyleTryon Do you think this would work?

@kelvintaywl, I'm usually up pretty late so feel free to Slack me and we can sync up

**Comment by brivu** [comment]
Hey @kelvintaywl, 

I took at look at your pipeline and wanted to know if you were you able to execute the `build-and-push` job with your changes? I see that you were able to run the `ecr-login` command successfully. If so I think your PR looks good. 

@KyleTryon, the implications are that if you pass the `role-arn`, `source-profile` and `new-profile` parameters into the `build-and-push` job, you're trying to run the job under the `new-profile`.

Otherwise, a user can configure the `role-arn` and run any of the `aws-ecr` commands in their own defined jobs.

-Brian

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:073a7986ad93b4828ee6c81c34dcb97483259dea

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:d4ddf2de0a69e7e7666190667982a2b57a9460cb

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:617a07a03d64623a2a01e1c7b27fa2e924d5aeea

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:7272e07c473b14dfc938a059332145a1925fcecf

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:eecc5503901034149f740c3ac335913bd40d6cf4

**Review by brivu — APPROVED** [review]
Looks really good.  Nice work @kelvintaywl

File: src/jobs/build-and-push-image.yml (hunk 1 above)
Comment (brivu) [reply src/jobs/build-and-push-image.yml:251]: good catch!

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:91d8a425fa32a0165e4d835563cc7d11aaa6803f

**Pull request merged**